projects/js-packages/publicize-components/src/components/unified-modal/edit-template/background-image-picker.tsx

```diff
@@ -0,0 +1,315 @@
+	featuredImageId: number | null,
+	defaultImageId: number | null
+): number | null {
+	switch ( imageType ) {
+		case 'default':
+			return defaultImageId;
+		case 'featured':
+			return featuredImageId;
+		case 'custom':
+			return customImageId;
+		case 'none':
+			return null;
+		default:
+			return featuredImageId;
+	}
+}
+
+/**
+ * BackgroundImagePicker component
+ *
+ * @param props                   - Component props
+ * @param props.imageType         - Current image type
+ * @param props.imageId           - Custom image ID
+ * @param props.defaultImageId    - Default image ID
+ * @param props.featuredImageId   - Featured image ID
+ * @param props.onImageTypeChange - Image type change handler
+ * @param props.onImageIdChange   - Image ID change handler
+ * @return BackgroundImagePicker component
+ */
+export function BackgroundImagePicker( {
+	imageType,
+	imageId,
+	defaultImageId,
+	featuredImageId,
+	onImageTypeChange,
+	onImageIdChange,
+}: BackgroundImagePickerProps ) {
+	// Ref to store the MediaUpload open function
+	const openMediaLibraryRef = useRef< () => void >( () => {} );
+
+	// Get the image ID to display
+	const displayImageId = useMemo(
+		() => getDisplayImageId( imageType, imageId, featuredImageId, defaultImageId ),
+		[ imageType, imageId, featuredImageId, defaultImageId ]
+	);
+
+	const [ mediaDetails ] = useMediaDetails( displayImageId );
+	const imageUrl = mediaDetails?.mediaData?.sourceUrl;
+	const isLoading = Boolean( displayImageId ) && ! imageUrl;
+	const showFeaturedImageNotice = imageType === 'featured' && ! featuredImageId;
+
+	// Build preview data for MediaPreview component
+	const previewData = useMemo( () => {
+		if ( ! imageUrl && ! isLoading ) {
+			return null;
+		}
+		return {
+			id: displayImageId || 0,
+			url: imageUrl || '',
+			type: 'image' as const,
+		};
+	}, [ imageUrl, isLoading, displayImageId ] );
+
+	// Build menu options (no "No Image" - Remove button handles that)
+	const menuOptions = useMemo( () => {
+		const options: MenuOption[] = [];
+
+		if ( defaultImageId ) {
+			options.push( {
+				id: 'default',
+				label: __( 'Default Image', 'jetpack-publicize-components' ),
+				icon: image,
+			} );
+		}
+
+		options.push( {
+			id: 'featured',
+			label: __( 'Featured Image', 'jetpack-publicize-components' ),
+			icon: image,
+		} );
+
+		options.push( {
+			id: 'custom',
+			label: __( 'Media Library', 'jetpack-publicize-components' ),
+			icon: mediaIcon,
+		} );
+
+		return options;
+	}, [ defaultImageId ] );
+
+	// Handle media library selection
+	const handleMediaSelect = useCallback(
+		( media: { id: number } ) => {
+			onImageTypeChange( 'custom' );
+			onImageIdChange( media?.id || null );
+		},
+		[ onImageTypeChange, onImageIdChange ]
+	);
+
+	const handleMediaLibraryClick = useCallback( () => {
+		setTimeout( () => {
+			openMediaLibraryRef.current();
+		}, 0 );
+	}, [] );
+
+	const renderMediaUpload = useCallback( ( { open }: { open: () => void } ) => {
```

Comment:
There is an alias for this

```suggestion
	const renderMediaUpload = useCallback( ( { open }: { open: VoidFunction } ) => {
```